<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color:  #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; margin-top: 20px; }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: grelightgreenen ; margin: 20px 0; }
    </style>
</head>
<body>

<h1>🏪 PROJET : "L'ARCHIPEL DES TERRASSES PARISIENNES"</h1>
<h2 style="text-align:center; color:lightgreen;">🔍 Analyse des clusters spontanés de terrasses commerciales à Paris</h2>
<hr>

<h2>🏪 1. Introduction</h2>
<p><strong>📌 Objectif :</strong></p>
<p>Ce projet vise à identifier et analyser les clusters spontanés de terrasses et étalages commerciaux à Paris.  
Comprendre ces patterns d'agglomération naturelle aide les urbanistes, investisseurs et commerçants à prendre des décisions éclairées.</p>

<h2>🛠️ Objectifs Clés</h2>
<ul>
    <li>🔄 Cartographier les clusters de terrasses par typologie</li>
    <li>📈 Analyser les corrélations spatiales entre établissements</li>
    <li>🏘️ Étudier l'influence des arrondissements sur la spécialisation</li>
    <li>🗺️ Identifier les "îles commerciales" extérieures spontanées</li>
</ul>

<h2>🔑 Définitions</h2>
<ul>
    <li>🏷️ <strong>Cluster Commercial :</strong> Regroupement naturel de terrasses similaires ou complémentaires. Exemples : zones de restaurants, corridors de cafés, concentrations d'étalages</li>
    <li>📊 <strong>Indice de Spécialisation :</strong> Concentration d'un type de terrasse dans une zone vs moyenne parisienne. Formule : (Type X terrasses / total terrasses en zone) × 100</li>
    <li>🌐 <strong>Écosystème Commercial Extérieur :</strong> Terrasses interagissant dans un espace, créant des synergies et complémentarités</li>
</ul>

<h2>🎯 Valeur Métier</h2>
<ul>
    <li>🏛️ Ville de Paris : Support décisionnel pour les autorisations de terrasses</li>
    <li>💼 Investisseurs : Identifier les zones sous-utilisées et opportunités de niche</li>
    <li>🛍️ Commerçants : Comprendre les écosystèmes locaux pour un positionnement optimal</li>
    <li>🏢 Urbanistes : Analyser les flux économiques et la vitalité des quartiers</li>
</ul>

<h2>👥 Parties Prenantes</h2>
<ul>
    <li>🏛️ Ville de Paris – Urbanisme et Développement Économique</li>
    <li>💼 Investisseurs Immobiliers – Promoteurs, sociétés foncières</li>
    <li>🛍️ Futurs Commerçants / Franchises</li>
    <li>🏢 Chambre de Commerce – Support aux entreprises</li>
    <li>🎓 Chercheurs en Économie Urbaine</li>
</ul>

<h2>📊 Critères de Succès</h2>
<ul>
    <li>🔍 Identifier au moins 5 clusters significatifs de terrasses</li>
    <li>📈 Mettre en lumière des corrélations spatiales contre-intuitives</li>
    <li>🎯 Produire des insights actionnables pour l'implantation commerciale</li>
    <li>📱 Créer des visualisations claires et pédagogiques</li>
</ul>

<h2>🚧 Contraintes</h2>
<ul>
    <li>📅 Délai : 5 jours</li>
    <li>💾 Données disponibles : 23,616 terrasses et étalages autorisés</li>
    <li>🔍 Qualité des données : Catégorisation des types de terrasses à uniformiser</li>
    <li>🌍 Périmètre : Paris intra-muros uniquement (20 arrondissements)</li>
    <li>✅ MVP : focus sur l'analyse spatiale des implantations extérieures</li>
</ul>

<h2>📋 Livrables</h2>
<ul>
    <li>📊 Rapport d'analyse complet avec méthodologie et résultats</li>
    <li>🗺️ Carte interactive des clusters identifiés</li>
    <li>📈 Tableau de bord des métriques clés</li>
    <li>💾 Code source et documentation technique</li>
    <li>🎤 Présentation synthétique pour les parties prenantes</li>
</ul>

</body>
</html>

<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; margin-top: 20px; }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>

<h1>🏪 2. Data Understanding</h2>
<p><strong>📌 Goal:</strong></p>
<p>The Data Understanding phase aims to explore and comprehend the dataset collected. This step ensures that all variables, their distributions, and the data quality are well understood before moving to analysis and modeling.</p>

<h2>🛠️ Key Elements of This Phase</h2>
<ul>
    <li>🔍 Dataset exploration: rows, columns, and types of variables</li>
    <li>🧹 Data quality assessment: missing values, inconsistencies, duplicates</li>
    <li>📊 Summary statistics: counts, averages, ranges, distributions</li>
    <li>🌐 Spatial analysis: examining locations, districts, coordinates</li>
    <li>🏷️ Categorical analysis: types of terraces, district classification, permit status</li>
    <li>📅 Temporal analysis: permit dates, seasonality, trends over time</li>
</ul>

<p>This phase sets the foundation for **data cleaning, preparation, and further analysis** by providing a clear understanding of what the dataset contains and its limitations.</p>


<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3>Imports</h3>


In [1]:
# %% Cellule 2: Installation et Import des Bibliothèques
print("📚 CHARGEMENT DES BIBLIOTHÈQUES...")

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.cluster import DBSCAN
from sklearn.preprocessing import StandardScaler
import folium
from folium.plugins import MarkerCluster
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots


print("✅ Bibliothèques chargées avec succès!")

📚 CHARGEMENT DES BIBLIOTHÈQUES...
✅ Bibliothèques chargées avec succès!


<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3>Data Loading</h3>


In [2]:

print("📂 CHARGEMENT DES DONNÉES...")

df_raw = pd.read_csv('../data/raw/terrasses_paris.csv')

print(f"✅ DONNÉES BRUTES CHARGÉES : {len(df_raw):,} terrasses")
print(f"📊 Dimensions : {df_raw.shape[0]} lignes x {df_raw.shape[1]} colonnes")


📂 CHARGEMENT DES DONNÉES...
✅ DONNÉES BRUTES CHARGÉES : 5,000 terrasses
📊 Dimensions : 5000 lignes x 12 colonnes


<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3>Exploration des colonnes brutes</h3>


In [3]:
print("\n👀 APERÇU DES DONNÉES BRUTES :")
display(df_raw.head(3))

print("\n📋 TOUTES LES COLONNES DISPONIBLES :")
for i, col in enumerate(df_raw.columns, 1):
    print(f"   {i:2d}. {col} - Type: {df_raw[col].dtype}")

print(f"\n🎯 COLONNES IMPORTANTES IDENTIFIÉES :")
important_cols = []
for col in df_raw.columns:
    if any(keyword in col.lower() for keyword in ['type', 'arrondissement', 'adresse', 'geo', 'nom', 'enseigne']):
        important_cols.append(col)
        print(f"   ✅ {col}")

for col in important_cols:
    print(f"\n📊 Aperçu de '{col}':")
    print(df_raw[col].head(5))


👀 APERÇU DES DONNÉES BRUTES :


typologie                 adresse  arrondissement  \
0  CONTRE-TERRASSE OUVERTE     14 RUE DES PANOYAUX         75020.0   
1         TERRASSE OUVERTE  167 BOULEVARD VOLTAIRE         75011.0   
2         TERRASSE OUVERTE        16 RUE DU LOUVRE         75001.0   

     nom_enseigne nom_societe         siret  longueur  largeur  \
0    LOU PASCALOU         NaN  5.238440e+13      4.80      4.0   
1             NaN         NaN  5.128560e+13      7.65      2.0   
2  CHEZ ALEXANDRE         NaN  4.222180e+13      2.50      1.1   

  periode_installation                                    lien_affichette  \
0                  NaN  https://eudonet-terrasses.apps.paris.fr/xrm/at...   
1                  NaN  https://eudonet-terrasses.apps.paris.fr/xrm/at...   
2                  NaN  https://eudonet-terrasses.apps.paris.fr/xrm/at...   

                                           geo_shape  \
0  {'type': 'Feature', 'geometry': {'coordinates'...   
1  {'type': 'Feature', 'geometry': {'coordinates'...   
2  {'type': 'Feature', 'geometry': {'coordinates'...   

                                        geo_point_2d  
0  {'lon': 2.3854522524772896, 'lat': 48.86633009...  
1  {'lon': 2.3857018993955266, 'lat': 48.85464237...  
2  {'lon': 2.3410809317051617, 'lat': 48.86116698...


📋 TOUTES LES COLONNES DISPONIBLES :
    1. typologie - Type: object
    2. adresse - Type: object
    3. arrondissement - Type: float64
    4. nom_enseigne - Type: object
    5. nom_societe - Type: object
    6. siret - Type: float64
    7. longueur - Type: float64
    8. largeur - Type: float64
    9. periode_installation - Type: object
   10. lien_affichette - Type: object
   11. geo_shape - Type: object
   12. geo_point_2d - Type: object

🎯 COLONNES IMPORTANTES IDENTIFIÉES :
   ✅ adresse
   ✅ arrondissement
   ✅ nom_enseigne
   ✅ nom_societe
   ✅ geo_shape
   ✅ geo_point_2d

📊 Aperçu de 'adresse':
0       14 RUE DES PANOYAUX
1    167 BOULEVARD VOLTAIRE
2          16 RUE DU LOUVRE
3       17 RUE LOUIS BONNET
4       65 RUE CAULAINCOURT
Name: adresse, dtype: object

📊 Aperçu de 'arrondissement':
0    75020.0
1    75011.0
2    75001.0
3    75011.0
4    75018.0
Name: arrondissement, dtype: float64

📊 Aperçu de 'nom_enseigne':
0              LOU PASCALOU
1                       NaN
2   

### 📌 Interpretation: Data Understanding

- **Columns Overview:** The dataset contains 12 columns, covering shop type, location, legal entity, dimensions, installation period, and geospatial info.
- **Key Columns Selected:** `adresse`, `arrondissement`, `nom_enseigne`, `nom_societe`, `geo_shape`, `geo_point_2d` — these are essential for spatial and business analysis.
- **Insights:**
  - `adresse` & `arrondissement` → help locate shops and analyze spatial distribution.
  - `nom_enseigne` & `nom_societe` → identify brand presence and company patterns.
  - `geo_shape` & `geo_point_2d` → allow mapping clusters and measuring proximity.


<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3>Qualité des colonnes brutes</h3>


In [4]:
print("\n🔍 ANALYSE DE LA QUALITÉ (BRUTES) :")

# Valeurs manquantes
missing_raw = df_raw.isnull().sum()
print("📊 VALEURS MANQUANTES PAR COLONNE :")
for col in df_raw.columns:
    if missing_raw[col] > 0:
        print(f"   ⚠️  {col:30} : {missing_raw[col]:6,} manquantes ({missing_raw[col]/len(df_raw)*100:5.1f}%)")

print(f"\n📈 STATISTIQUES DE QUALITÉ :")
print(f"   • Taux de complétude global : {(1 - missing_raw.sum() / (len(df_raw) * len(df_raw.columns))) * 100:.1f}%")
print(f"   • Colonnes sans données manquantes : {sum(missing_raw == 0)}/{len(df_raw.columns)}")
print(f"   • Colonnes avec +20% de données manquantes : {sum(missing_percent > 20)}")



# Graphique des valeurs manquantes
fig = px.bar(x=missing_data.index, y=missing_percent, 
             title='📊 POURCENTAGE DE VALEURS MANQUANTES PAR COLONNE',
             labels={'x': 'Colonnes', 'y': '% Valeurs Manquantes'},
             color=missing_percent,
             color_continuous_scale='reds')

fig.update_layout(xaxis_tickangle=-45, showlegend=False)
fig.show()

# Métriques de qualité
completeness = (1 - df_raw.isnull().sum().sum() / (len(df_raw) * len(df_raw.columns))) * 100



🔍 ANALYSE DE LA QUALITÉ (BRUTES) :
📊 VALEURS MANQUANTES PAR COLONNE :
   ⚠️  typologie                      :      2 manquantes (  0.0%)
   ⚠️  adresse                        :      2 manquantes (  0.0%)
   ⚠️  arrondissement                 :    150 manquantes (  3.0%)
   ⚠️  nom_enseigne                   :    155 manquantes (  3.1%)
   ⚠️  nom_societe                    :  4,847 manquantes ( 96.9%)
   ⚠️  siret                          :    175 manquantes (  3.5%)
   ⚠️  longueur                       :     16 manquantes (  0.3%)
   ⚠️  largeur                        :     24 manquantes (  0.5%)
   ⚠️  periode_installation           :  4,848 manquantes ( 97.0%)
   ⚠️  lien_affichette                :    923 manquantes ( 18.5%)

📈 STATISTIQUES DE QUALITÉ :
   • Taux de complétude global : 81.4%
   • Colonnes sans données manquantes : 2/12


NameError: name 'missing_percent' is not defined

### 🔍 Interpretation: Data Quality

- **Missing Values Overview:**  
  - Most columns have very few missing values (`typologie`, `adresse`, `arrondissement`, `nom_enseigne`, `siret`, `longueur`, `largeur`).  
  - Columns `nom_societe` and `periode_installation` have extremely high missing rates (~97%), indicating they may not be reliable for analysis.  
  - `lien_affichette` has 18.5% missing values — could be used selectively.  

- **Global Quality Metrics:**  
  - Overall completeness: 81.4%  
  - Only 2 columns are fully complete.  
  - 2 columns have more than 20% missing values, signaling caution or potential exclusion.  



<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3>Analyse des données géographiques brutes</h3>


In [ ]:
print("\n🗺️  ANALYSE DES DONNÉES GÉOGRAPHIQUES :")

geo_cols = [col for col in df_raw.columns if any(keyword in col.lower() for keyword in ['geo', 'lat', 'lon', 'coord'])]
print(f"📍 Colonnes géographiques trouvées : {geo_cols}")

for geo_col in geo_cols:
    print(f"\n📊 Analyse de '{geo_col}':")
    print(f"   • Type : {df_raw[geo_col].dtype}")
    print(f"   • Valeurs uniques : {df_raw[geo_col].nunique()}")
    print(f"   • Exemples : {df_raw[geo_col].dropna().head(3).tolist()}")
    
    sample_val = df_raw[geo_col].dropna().iloc[0] if not df_raw[geo_col].dropna().empty else "N/A"
    if isinstance(sample_val, str) and ',' in sample_val:
        print(f"   • Format : Coordonnées combinées (lat,lon)")
    else:
        print(f"   • Format : Simple")






🗺️  ANALYSE DES DONNÉES GÉOGRAPHIQUES :
📍 Colonnes géographiques trouvées : ['longueur', 'periode_installation', 'geo_shape', 'geo_point_2d']

📊 Analyse de 'longueur':
   • Type : float64
   • Valeurs uniques : 712
   • Exemples : [4.8, 7.65, 2.5]
   • Format : Simple

📊 Analyse de 'periode_installation':
   • Type : object
   • Valeurs uniques : 1
   • Exemples : ["Toute l'année", "Toute l'année", "Toute l'année"]
   • Format : Simple

📊 Analyse de 'geo_shape':
   • Type : object
   • Valeurs uniques : 4988
   • Exemples : ["{'type': 'Feature', 'geometry': {'coordinates': [2.3854522524772896, 48.866330091701045], 'type': 'Point'}, 'properties': {}}", "{'type': 'Feature', 'geometry': {'coordinates': [2.3857018993955266, 48.854642373630554], 'type': 'Point'}, 'properties': {}}", "{'type': 'Feature', 'geometry': {'coordinates': [2.3410809317051617, 48.86116698012297], 'type': 'Point'}, 'properties': {}}"]
   • Format : Coordonnées combinées (lat,lon)

📊 Analyse de 'geo_point_2d':
   • T

### 🌍 Interpretation: Geographical Data

- **Columns Identified:** `longueur`, `periode_installation`, `geo_shape`, `geo_point_2d`  

- **`longueur`:**  
  - Numeric and simple, with 712 unique values.  
  - Can be used to analyze terrace lengths and spatial distribution.  

- **`periode_installation`:**  
  - Single repeated value ("Toute l'année"), so this column has no variability.  
  - Not useful for analysis of seasonal patterns.  

- **`geo_shape`:**  
  - Contains 4988 unique coordinates in a complex object format.  
  - Represents precise locations of terraces; can be used for mapping and spatial clustering.  

- **`geo_point_2d`:**  
  - Also 4988 unique coordinates in a simplified latitude/longitude format.  
  - Easier to use for visualization, geospatial computations, and cluster analysis.  




<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3>Analyse des types et catégories brutes</h3>


In [ ]:
print("\n🏷️  ANALYSE DES CATÉGORIES ET TYPES :")

type_cols = [col for col in df_raw.columns if any(keyword in col.lower() for keyword in ['type', 'typologie', 'categorie', 'catégorie'])]
print(f"🎪 Colonnes de type trouvées : {type_cols}")

for type_col in type_cols:
    print(f"\n📊 Analyse de '{type_col}':")
    type_stats = df_raw[type_col].value_counts()
    print(f"   • Valeurs uniques : {type_stats.nunique()}")
    print(f"   • Top 5 valeurs :")
    for i, (val, count) in enumerate(type_stats.head().items(), 1):
        print(f"      {i}. {val} : {count:,}")


🏷️  ANALYSE DES CATÉGORIES ET TYPES :
🎪 Colonnes de type trouvées : ['typologie']

📊 Analyse de 'typologie':
   • Valeurs uniques : 24
   • Top 5 valeurs :
      1. TERRASSE OUVERTE : 1,892
      2. ETALAGE : 985
      3. CONTRE TERRASSE ESTIVALE SUR STATIONNEMENT : 464
      4. TERRASSE FERMÉE : 408
      5. TERRASSES OUVERTES SUR TROTTOIR : 346


### 🏷️ Interpretation: Categories & Types

- **Column:** `typologie`  
- **Unique Values:** 24  
- **Most Frequent Types:**  
  1. **TERRASSE OUVERTE** – 1,892 occurrences  
  2. **ETALAGE** – 985 occurrences  
  3. **CONTRE TERRASSE ESTIVALE SUR STATIONNEMENT** – 464 occurrences  
  4. **TERRASSE FERMÉE** – 408 occurrences  
  5. **TERRASSES OUVERTES SUR TROTTOIR** – 346 occurrences  

- **Insights:**  
  - Open terraces ("TERRASSE OUVERTE") dominate the dataset, showing a strong outdoor commercial presence.  
  - Displays/étalages are also common, indicating commercial activity on sidewalks.  
  - Some terrace types are more specialized (e.g., "CONTRE TERRASSE ESTIVALE"), which could be interesting for spatial clustering.  




<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3>Analyse des arronsissements brutes</h3>


In [ ]:
# Cellule 4: ANALYSE INTERACTIVE DES ARRONDISSEMENTS
print("🏙️ ANALYSE INTERACTIVE DES ARRONDISSEMENTS :")

# Recherche de la colonne d'arrondissement
arr_cols = [col for col in df_raw.columns if any(keyword in col.lower() for keyword in ['arrondissement', 'arrdt', 'arr'])]

if arr_cols:
    arr_col = arr_cols[0]  # Prendre la première colonne trouvée
    print(f"📍 Utilisation de la colonne : {arr_col}")
    
    arr_stats = df_raw[arr_col].value_counts().reset_index()
    arr_stats.columns = ['Arrondissement', 'Nombre_Terrasses']
    
    # Graphique en barres interactif
    fig = px.bar(arr_stats, x='Arrondissement', y='Nombre_Terrasses',
                 title='🏙️ NOMBRE DE TERRASSES PAR ARRONDISSEMENT',
                 color='Nombre_Terrasses',
                 color_continuous_scale='viridis')
    
    fig.update_layout(xaxis_tickangle=-45)
    fig.show()
    
    # Camembert pour la répartition
    fig_pie = px.pie(arr_stats.head(10), names='Arrondissement', values='Nombre_Terrasses',
                     title='🎪 TOP 10 DES ARRONDISSEMENTS (RÉPARTITION)')
    fig_pie.show()
    
    print(f"📊 STATISTIQUES :")
    print(f"   • {arr_stats['Arrondissement'].nunique()} arrondissements différents")
    print(f"   • Moyenne : {arr_stats['Nombre_Terrasses'].mean():.0f} terrasses/arrondissement")
    print(f"   • Max : {arr_stats['Nombre_Terrasses'].max()} terrasses")
    
else:
    print("❌ Aucune colonne d'arrondissement trouvée")

🏙️ ANALYSE INTERACTIVE DES ARRONDISSEMENTS :
📍 Utilisation de la colonne : arrondissement


📊 STATISTIQUES :
   • 21 arrondissements différents
   • Moyenne : 231 terrasses/arrondissement
   • Max : 341 terrasses


### 🏙️ Interpretation: Arrondissements Analysis

- **Column Used:** `arrondissement`  
- **Number of Districts:** 21  
- **Average Terraces per District:** 231  
- **Maximum Terraces in a District:** 341  

- **Insights:**  
  - Terraces are unevenly distributed across Paris, with some arrondissements showing much higher concentrations.  
  - The top 10 districts account for the majority of terrace activity, which may indicate commercial hotspots.  
  - This distribution can guide targeted urban planning and commercial positioning strategies.  



<!DOCTYPE html>
<html>
<head>
    <style>
        body { 
            font-family: Arial, sans-serif; 
            margin: 40px; 
            background: #f5f5f5; 
            color: #333; 
            line-height: 1.6; 
        }
        h1 { 
            text-align: center; 
            color: lightgreen; 
            font-size: 36px; 
            font-weight: bold; 
            margin-bottom: 10px; 
        }
        h2 { 
            color: #EF806C; 
            font-size: 28px; 
            font-weight: bold; 
            margin-top: 30px; 
        }
        h3 { 
            color: #00FFFF; 
            font-size: 22px; 
            margin-top: 20px; 
        }
        ul { 
            margin-left: 20px; 
        }
        li { 
            margin: 8px 0; 
        }
        hr { 
            height: 2px; 
            border: none; 
            background-color: lightgreen; 
            margin: 20px 0; 
        }
    </style>
</head>
<body>

<h1>🧹 3. Data Preparation</h1>
<p><strong>📌 Goal:</strong></p>
<p>The Data Preparation phase focuses on transforming raw data into a clean and structured format suitable for analysis. This stage ensures that the dataset is consistent, complete, and ready for visualization or modeling.</p>

<h2>🛠️ Key Elements of This Phase</h2>
<ul>
    <li>🧩 Handling missing data: filling, imputing, or removing incomplete records</li>
    <li>🧽 Data cleaning: correcting errors, standardizing text, removing duplicates</li>
    <li>🔢 Data transformation: converting data types, normalizing numerical fields</li>
    <li>🏷️ Feature selection: retaining relevant variables for analysis</li>
    <li>🗺️ Spatial data formatting: ensuring coordinate consistency (lat/lon)</li>
    <li>🧮 Creating derived variables: extracting insights from existing features</li>
</ul>

<p>This step bridges raw exploration and advanced analytics by ensuring the data is accurate, usable, and aligned with project goals.</p>


</body>
</html>



<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3>Création de la copie de sécurité</h3>


In [5]:

print("🔧 ÉTAPE 1: CRÉATION DE LA COPIE DE SÉCURITÉ")

df_clean = df_raw.copy()


print(f"\n📊 AVANT NETTOYAGE : {len(df_clean):,} enregistrements")
print(f"📋 Colonnes disponibles : {len(df_clean.columns)}")

🔧 ÉTAPE 1: CRÉATION DE LA COPIE DE SÉCURITÉ

📊 AVANT NETTOYAGE : 5,000 enregistrements
📋 Colonnes disponibles : 12


<h2>🎯 Conclusion</h2>
<ul>
    <li>💾 <strong>Preserve raw data:</strong> Maintaining the original dataset ensures data integrity and allows future reference or verification.</li>
    <li>🧩 <strong>Work on a copy:</strong> Performing cleaning and transformations on a duplicate prevents accidental data loss or corruption.</li>
    <li>⚖️ <strong>Enable before/after comparison:</strong> Keeping both versions (raw and prepared) allows validation of data quality improvements and reproducibility of results.</li>
</ul>

<p>➡️ In short, this step builds a reliable, structured dataset — the foundation for accurate analysis and meaningful insights in the next phase.</p>



<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3> Extraction des coordonnées GPS</h3>


In [6]:

print("📍 ÉTAPE 2: EXTRACTION DES COORDONNÉES GPS")

def extract_coordinates(geo_str):
    """
    Extrait latitude et longitude de la chaîne JSON geo_point_2d
    """
    try:
        if pd.isna(geo_str):
            return np.nan, np.nan
        
        # Convertir la string JSON en dictionnaire Python
        import ast
        geo_dict = ast.literal_eval(geo_str)
        
        # Extraire lat et lon
        lat = geo_dict.get('lat', np.nan)
        lon = geo_dict.get('lon', np.nan)
        
        return lat, lon
        
    except Exception as e:
        print(f"⚠️  Erreur d'extraction : {e}")
        return np.nan, np.nan

# Appliquer l'extraction à toutes les lignes
print("🔄 Extraction en cours...")
coordinates_data = df_clean['geo_point_2d'].apply(
    lambda x: pd.Series(extract_coordinates(x)) if pd.notna(x) else pd.Series([np.nan, np.nan])
)

# Ajouter les nouvelles colonnes
df_clean[['latitude', 'longitude']] = coordinates_data

print(f"\n✅ RÉSULTATS :")
print(f"   • Points géolocalisés extraits : {df_clean[['latitude', 'longitude']].notna().all(axis=1).sum():,}")
print(f"   • Latitude min/max : {df_clean['latitude'].min():.3f} / {df_clean['latitude'].max():.3f}")
print(f"   • Longitude min/max : {df_clean['longitude'].min():.3f} / {df_clean['longitude'].max():.3f}")

# Aperçu des coordonnées extraites
print("\n👀 APERÇU DES COORDONNÉES EXTRAITES :")
display(df_clean[['geo_point_2d', 'latitude', 'longitude']].head(3))

📍 ÉTAPE 2: EXTRACTION DES COORDONNÉES GPS
🔄 Extraction en cours...

✅ RÉSULTATS :
   • Points géolocalisés extraits : 5,000
   • Latitude min/max : 48.819 / 48.899
   • Longitude min/max : 2.255 / 2.411

👀 APERÇU DES COORDONNÉES EXTRAITES :


geo_point_2d   latitude  longitude
0  {'lon': 2.3854522524772896, 'lat': 48.86633009...  48.866330   2.385452
1  {'lon': 2.3857018993955266, 'lat': 48.85464237...  48.854642   2.385702
2  {'lon': 2.3410809317051617, 'lat': 48.86116698...  48.861167   2.341081

<h2>📍 cocnlusion </h2>

<h3>🎯 Objectif</h3>
<ul>
    <li>🗺️ <strong>Format JSON à décoder :</strong> Les coordonnées géographiques sont stockées dans un champ JSON (<code>geo_point_2d</code>), nécessitant une extraction manuelle.</li>
    <li>📍 <strong>Séparation des valeurs :</strong> Il est essentiel de dissocier la <strong>latitude</strong> et la <strong>longitude</strong> en deux colonnes distinctes pour l'analyse géospatiale.</li>
    <li>🤖 <strong>Préparation pour le clustering :</strong> L’algorithme <code>DBSCAN</code> requiert des variables numériques indépendantes afin d’identifier les regroupements spatiaux (clusters).</li>
</ul>

<p>✅ Cette étape garantit la bonne préparation des données géographiques avant toute modélisation spatiale et permet d’obtenir des résultats de clustering précis et exploitables.</p>



<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3> Nettoyage des arrondissements</h3>


In [9]:

print("🏙️ ÉTAPE 3: NETTOYAGE DES ARRONDISSEMENTS")

# Conversion et nettoyage
print("🔄 Conversion des arrondissements...")

# Sauvegarde du type original
original_dtype = df_clean['arrondissement'].dtype
original_nunique = df_clean['arrondissement'].nunique()

# Conversion en string et nettoyage
df_clean['arrondissement'] = df_clean['arrondissement'].astype(str)
df_clean['arrondissement'] = df_clean['arrondissement'].str.replace('.0', '', regex=False)
df_clean['arrondissement'] = df_clean['arrondissement'].replace('nan', np.nan)

print(f"\n✅ RÉSULTATS :")
print(f"   • Type avant : {original_dtype}")
print(f"   • Type après : {df_clean['arrondissement'].dtype}")
print(f"   • Arrondissements uniques : {original_nunique} → {df_clean['arrondissement'].nunique()}")
print(f"   • Format standardisé : '75020' au lieu de '75020.0'")

# Distribution des arrondissements
print("\n📊 DISTRIBUTION DES ARRONDISSEMENTS (après nettoyage) :")
arrondissement_stats = df_clean['arrondissement'].value_counts().head(10)
for i, (arr, count) in enumerate(arrondissement_stats.items(), 1):
    print(f"   {i:2d}. {arr} : {count:,} terrasses")

🏙️ ÉTAPE 3: NETTOYAGE DES ARRONDISSEMENTS
🔄 Conversion des arrondissements...

✅ RÉSULTATS :
   • Type avant : object
   • Type après : object
   • Arrondissements uniques : 21 → 21
   • Format standardisé : '75020' au lieu de '75020.0'

📊 DISTRIBUTION DES ARRONDISSEMENTS (après nettoyage) :
    1. 75018 : 341 terrasses
    2. 75015 : 335 terrasses
    3. 75011 : 333 terrasses
    4. 75010 : 322 terrasses
    5. 75017 : 299 terrasses
    6. 75014 : 290 terrasses
    7. 75005 : 289 terrasses
    8. 75012 : 275 terrasses
    9. 75008 : 260 terrasses
   10. 75009 : 251 terrasses


<h2>🏙️ Conclusion</h2>

<h3>🎯 Objectif</h3>
<ul>
    <li>🔢 <strong>Problème de type :</strong> Les valeurs des arrondissements sont enregistrées au format <code>float</code> (ex. 75020.0), ce qui n’est pas adapté à une analyse catégorielle.</li>
    <li>🔤 <strong>Conversion nécessaire :</strong> Transformation en <code>string</code> pour permettre les regroupements, tris et visualisations correctes.</li>
    <li>🧩 <strong>Uniformisation du format :</strong> Assure la cohérence entre les observations et facilite les comparaisons inter-arrondissements.</li>
</ul>

<p>✅ Cette étape simplifie l’analyse statistique et visuelle, garantissant une catégorisation claire des arrondissements lors des analyses ultérieures.</p>



<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3> Standardisation des typologies </h3>


In [10]:

print("🏷️ ÉTAPE 4: STANDARDISATION DES TYPOLOGIES")

# Sauvegarde des stats avant nettoyage
original_typologies = df_clean['typologie'].nunique()
original_sample = df_clean['typologie'].head(5).tolist()

print("🔄 Standardisation en cours...")

# Nettoyage des typologies
df_clean['typologie'] = df_clean['typologie'].str.strip().str.upper()
df_clean['typologie'] = df_clean['typologie'].replace('NAN', np.nan)

print(f"\n✅ RÉSULTATS :")
print(f"   • Types uniques avant : {original_typologies}")
print(f"   • Types uniques après : {df_clean['typologie'].nunique()}")
print(f"   • Exemple avant : {original_sample}")
print(f"   • Exemple après : {df_clean['typologie'].head(5).tolist()}")

# Top des typologies après nettoyage
print("\n📊 TOP 10 DES TYPOLOGIES (standardisées) :")
typologie_stats = df_clean['typologie'].value_counts().head(10)
for i, (typo, count) in enumerate(typologie_stats.items(), 1):
    print(f"   {i:2d}. {typo} : {count:,}")

🏷️ ÉTAPE 4: STANDARDISATION DES TYPOLOGIES
🔄 Standardisation en cours...

✅ RÉSULTATS :
   • Types uniques avant : 32
   • Types uniques après : 32
   • Exemple avant : ['CONTRE-TERRASSE OUVERTE', 'TERRASSE OUVERTE', 'TERRASSE OUVERTE', 'ETALAGE', 'TERRASSE OUVERTE']
   • Exemple après : ['CONTRE-TERRASSE OUVERTE', 'TERRASSE OUVERTE', 'TERRASSE OUVERTE', 'ETALAGE', 'TERRASSE OUVERTE']

📊 TOP 10 DES TYPOLOGIES (standardisées) :
    1. TERRASSE OUVERTE : 1,892
    2. ETALAGE : 985
    3. CONTRE TERRASSE ESTIVALE SUR STATIONNEMENT : 464
    4. TERRASSE FERMÉE : 408
    5. TERRASSES OUVERTES SUR TROTTOIR : 346
    6. TERRASSE ESTIVALE SUR TROTTOIR FACE À LA DEVANTURE : 225
    7. CONTRE TERRASSE ESTIVALE SUR TROTTOIR FACE À LA DEVANTURE : 159
    8. CONTRE-TERRASSE PERMANENTE SUR STATIONNEMENT : 59
    9. ÉTALAGE SUR TROTTOIR : 59
   10. COMMERCE ACCESSOIRE : 57


<h2>🧹Conclusion</h2>

<h3>🎯 Objectif</h3>
<ul>
    <li>🔠 <strong>Uniformiser la casse :</strong> Convertir toutes les valeurs en minuscules (ou majuscules) afin d’éviter les doublons dus aux variations de casse.</li>
    <li>✂️ <strong>Supprimer les espaces superflus :</strong> Éliminer les espaces en début ou fin de chaîne pour garantir la cohérence des données textuelles.</li>
    <li>🏷️ <strong>Préparer les catégories :</strong> Faciliter les regroupements, tris et visualisations lors des futures analyses catégorielles.</li>
</ul>

<p>✅ Cette étape améliore la qualité des données textuelles et garantit des regroupements fiables lors de l’analyse exploratoire.</p>



<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3> Filtrage des données complètes </h3>


In [11]:

print("🎯 ÉTAPE 5: FILTRAGE DES DONNÉES COMPLÈTES")

initial_count = len(df_clean)
print(f"📊 Avant filtrage : {initial_count:,} enregistrements")

# Colonnes critiques pour l'analyse
critical_columns = ['latitude', 'longitude', 'typologie']

print(f"\n🔍 VÉRIFICATION DES DONNÉES MANQUANTES :")
for col in critical_columns:
    missing = df_clean[col].isna().sum()
    print(f"   • {col:15} : {missing:5,} manquantes ({missing/initial_count*100:5.1f}%)")

# Filtrage
print("\n🔄 Filtrage en cours...")
df_clean = df_clean.dropna(subset=critical_columns)
final_count = len(df_clean)

print(f"\n✅ RÉSULTATS :")
print(f"   • Enregistrements conservés : {final_count:,}")
print(f"   • Taux de conservation : {final_count/initial_count*100:.1f}%")
print(f"   • Données éliminées : {initial_count - final_count:,}")

🎯 ÉTAPE 5: FILTRAGE DES DONNÉES COMPLÈTES
📊 Avant filtrage : 5,000 enregistrements

🔍 VÉRIFICATION DES DONNÉES MANQUANTES :
   • latitude        :     0 manquantes (  0.0%)
   • longitude       :     0 manquantes (  0.0%)
   • typologie       :     2 manquantes (  0.0%)

🔄 Filtrage en cours...

✅ RÉSULTATS :
   • Enregistrements conservés : 4,998
   • Taux de conservation : 100.0%
   • Données éliminées : 2


<h2>🧭 Conclusion </h2>

<h3>🎯 Objectif</h3>
<ul>
    <li>🧹 <strong>Supprimer les lignes incomplètes :</strong> Retirer les enregistrements contenant des valeurs manquantes sur des champs essentiels à l’analyse.</li>
    <li>📍 <strong>Conserver uniquement les données géolocalisées :</strong> Garder les lignes avec des coordonnées GPS et une typologie valide pour permettre les analyses spatiales.</li>
    <li>📊 <strong>Améliorer la fiabilité des analyses :</strong> Garantir que les visualisations et modèles reposent sur des données complètes et cohérentes.</li>
</ul>

<p>✅ Cette étape assure une base de données propre, exploitable et prête pour les étapes de clustering et d’exploration géographique.</p>



<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3> Validation des donées geographique  </h3>


In [12]:

print("🗺️ ÉTAPE 6: VALIDATION GÉOGRAPHIQUE")


# Bornes géographiques de Paris intra-muros
paris_bounds = {
    'lat_min': 48.81, 'lat_max': 48.91,   # Latitude Nord/Sud
    'lon_min': 2.25, 'lon_max': 2.41      # Longitude Est/Ouest
}

print(f"📍 BORNES DE PARIS INTRA-MUROS :")
print(f"   • Latitude  : {paris_bounds['lat_min']:.3f} → {paris_bounds['lat_max']:.3f}")
print(f"   • Longitude : {paris_bounds['lon_min']:.3f} → {paris_bounds['lon_max']:.3f}")

initial_geo_count = len(df_clean)

# Application du filtre géographique
mask_paris = (
    (df_clean['latitude'] >= paris_bounds['lat_min']) & 
    (df_clean['latitude'] <= paris_bounds['lat_max']) &
    (df_clean['longitude'] >= paris_bounds['lon_min']) & 
    (df_clean['longitude'] <= paris_bounds['lon_max'])
)

df_clean = df_clean[mask_paris]
final_geo_count = len(df_clean)

print(f"\n✅ RÉSULTATS :")
print(f"   • Points avant validation : {initial_geo_count:,}")
print(f"   • Points après validation : {final_geo_count:,}")
print(f"   • Points éliminés (hors Paris) : {initial_geo_count - final_geo_count:,}")

# Vérification des coordonnées
print(f"\n🌍 COORDONNÉES FINALES :")
print(f"   • Latitude min/max  : {df_clean['latitude'].min():.3f} / {df_clean['latitude'].max():.3f}")
print(f"   • Longitude min/max : {df_clean['longitude'].min():.3f} / {df_clean['longitude'].max():.3f}")

🗺️ ÉTAPE 6: VALIDATION GÉOGRAPHIQUE
📍 BORNES DE PARIS INTRA-MUROS :
   • Latitude  : 48.810 → 48.910
   • Longitude : 2.250 → 2.410

✅ RÉSULTATS :
   • Points avant validation : 4,998
   • Points après validation : 4,988
   • Points éliminés (hors Paris) : 10

🌍 COORDONNÉES FINALES :
   • Latitude min/max  : 48.819 / 48.899
   • Longitude min/max : 2.255 / 2.410


<h2>🗺️ Conclusion</h2>

<h3>🎯 Objectif</h3>
<ul>
    <li>📍 <strong>Vérifier la localisation :</strong> S’assurer que toutes les coordonnées GPS appartiennent bien au territoire parisien (intra-muros).</li>
    <li>🚫 <strong>Éliminer les anomalies géographiques :</strong> Supprimer les points erronés, situés en dehors de Paris ou mal enregistrés.</li>
    <li>🧭 <strong>Garantir la cohérence spatiale :</strong> Préparer un jeu de données géographiquement homogène et fiable pour l’analyse des clusters.</li>
</ul>

<p>✅ Ce filtrage renforce la précision des résultats et évite les biais lors du calcul des regroupements commerciaux ou des visualisations cartographiques.</p>


In [13]:

print("💾 ÉTAPE 7: SAUVEGARDE ET RÉSUMÉ FINAL")

# Résumé complet du nettoyage
print("\n" + "="*70)
print("🎉 RÉSUMÉ COMPLET DU NETTOYAGE - PHASE 3 TERMINÉE !")
print("="*70)

cleaning_summary = {
    "Enregistrements bruts initiaux": len(df_raw),
    "Enregistrements après nettoyage": len(df_clean),
    "Taux de conservation global": f"{(len(df_clean)/len(df_raw)*100):.1f}%",
    "Points géolocalisés valides": df_clean[['latitude', 'longitude']].notna().all(axis=1).sum(),
    "Arrondissements couverts": df_clean['arrondissement'].nunique(),
    "Types de terrasses": df_clean['typologie'].nunique(),
    "Données dans Paris intra-muros": "100%"
}

print("\n📊 STATISTIQUES FINALES :")
for key, value in cleaning_summary.items():
    if isinstance(value, str):
        print(f"   • {key:35} : {value}")
    else:
        print(f"   • {key:35} : {value:,}")

# Sauvegarde des données nettoyées
print("\n💾 SAUVEGARDE DES DONNÉES NETTOYÉES...")
df_clean.to_csv('../data/processed/terrasses_paris_nettoyees.csv', index=False, encoding='utf-8')
print("✅ Données sauvegardées : 'data/processed/terrasses_paris_nettoyees.csv'")

print("\n👀 APERÇU FINAL DES DONNÉES NETTOYÉES :")
display(df_clean[['typologie', 'arrondissement', 'latitude', 'longitude', 'adresse']].head(5))

print("\n🎯 PROCHAINE ÉTAPE : PHASE 4 - MODELING (Clustering spatial)")
print("➡️  Prêt pour l'analyse des clusters avec DBSCAN!")

💾 ÉTAPE 7: SAUVEGARDE ET RÉSUMÉ FINAL

🎉 RÉSUMÉ COMPLET DU NETTOYAGE - PHASE 3 TERMINÉE !

📊 STATISTIQUES FINALES :
   • Enregistrements bruts initiaux      : 5,000
   • Enregistrements après nettoyage     : 4,988
   • Taux de conservation global         : 99.8%
   • Points géolocalisés valides         : 4,988
   • Arrondissements couverts            : 21
   • Types de terrasses                  : 32
   • Données dans Paris intra-muros      : 100%

💾 SAUVEGARDE DES DONNÉES NETTOYÉES...
✅ Données sauvegardées : 'data/processed/terrasses_paris_nettoyees.csv'

👀 APERÇU FINAL DES DONNÉES NETTOYÉES :


typologie arrondissement   latitude  longitude  \
0  CONTRE-TERRASSE OUVERTE          75020  48.866330   2.385452   
1         TERRASSE OUVERTE          75011  48.854642   2.385702   
2         TERRASSE OUVERTE          75001  48.861167   2.341081   
3                  ETALAGE          75011  48.871256   2.376613   
4         TERRASSE OUVERTE          75018  48.889594   2.335879   

                  adresse  
0     14 RUE DES PANOYAUX  
1  167 BOULEVARD VOLTAIRE  
2        16 RUE DU LOUVRE  
3     17 RUE LOUIS BONNET  
4     65 RUE CAULAINCOURT


🎯 PROCHAINE ÉTAPE : PHASE 4 - MODELING (Clustering spatial)
➡️  Prêt pour l'analyse des clusters avec DBSCAN!



<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3> Verification des types de données </h3>


In [15]:

print("🔍 TEST 1: VÉRIFICATION DES TYPES DE DONNÉES")


data_types_check = {
    'latitude': 'float64',
    'longitude': 'float64', 
    'typologie': 'object',
    'arrondissement': 'object'
}

print("\n📊 TYPES ACTUELS DES COLONNES CLÉS :")
for col, expected_type in data_types_check.items():
    actual_type = df_clean[col].dtype
    status = "✅" if str(actual_type) == expected_type else "❌"
    print(f"   {status} {col:15} : {actual_type} (attendu: {expected_type})")

# Vérification supplémentaire pour les coordonnées
print(f"\n📍 VÉRIFICATION NUMÉRIQUE DES COORDONNÉES :")
print(f"   • Latitude  : Min {df_clean['latitude'].min():.6f}, Max {df_clean['latitude'].max():.6f}")
print(f"   • Longitude : Min {df_clean['longitude'].min():.6f}, Max {df_clean['longitude'].max():.6f}")
print(f"   • Valeurs NaN : Lat {df_clean['latitude'].isna().sum()}, Lon {df_clean['longitude'].isna().sum()}")

🔍 TEST 1: VÉRIFICATION DES TYPES DE DONNÉES

📊 TYPES ACTUELS DES COLONNES CLÉS :
   ✅ latitude        : float64 (attendu: float64)
   ✅ longitude       : float64 (attendu: float64)
   ✅ typologie       : object (attendu: object)
   ✅ arrondissement  : object (attendu: object)

📍 VÉRIFICATION NUMÉRIQUE DES COORDONNÉES :
   • Latitude  : Min 48.819138, Max 48.898886
   • Longitude : Min 2.255028, Max 2.409811
   • Valeurs NaN : Lat 0, Lon 0


<h2>🧪 Test 1 : Conclusion</h2>

<h3>🎯 Objectif</h3>
<p>S’assurer que chaque colonne possède le bon type de données afin de garantir la fiabilité des traitements et analyses ultérieurs.</p>

<h3>📊 Points de Contrôle</h3>
<ul>
    <li>✅ <strong>Latitude</strong> → Doit être de type <code>float64</code></li>
    <li>✅ <strong>Longitude</strong> → Doit être de type <code>float64</code></li>
    <li>✅ <strong>Typologie</strong> → Doit être de type <code>object</code></li>
    <li>✅ <strong>Arrondissement</strong> → Doit être de type <code>object</code></li>
</ul>

<h3>📍 Vérification Numérique des Coordonnées</h3>
<ul>
    <li>📈 <strong>Latitude :</strong> bornes min/max conformes à la zone parisienne (≈ 48.80 à 48.90)</li>
    <li>📉 <strong>Longitude :</strong> bornes min/max conformes (≈ 2.25 à 2.42)</li>
    <li>⚙️ <strong>Valeurs manquantes :</strong> Absence ou faible nombre de NaN validée</li>
</ul>

<p>✅ Cette étape confirme la conformité structurelle du jeu de données avant le passage aux étapes de clustering et de cartographie.</p>


<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3> Verification des valeurs manquantes  </h3>


In [16]:

print("🔍 TEST 2: VÉRIFICATION DES VALEURS MANQUANTES")


critical_columns = ['latitude', 'longitude', 'typologie', 'arrondissement']

print("\n📊 VALEURS MANQUANTES PAR COLONNE CRITIQUE :")
missing_summary = []
all_critical_ok = True

for col in critical_columns:
    missing_count = df_clean[col].isna().sum()
    missing_percent = (missing_count / len(df_clean)) * 100
    status = "✅" if missing_count == 0 else "❌"
    
    if missing_count > 0:
        all_critical_ok = False
        
    missing_summary.append({
        'Colonne': col, 
        'Manquantes': missing_count,
        'Pourcentage': f"{missing_percent:.2f}%",
        'Status': status
    })
    
    print(f"   {status} {col:15} : {missing_count:4} manquantes ({missing_percent:.2f}%)")

print(f"\n🎯 RÉSULTAT DU TEST : {'✅ TOUT EST COMPLET' if all_critical_ok else '❌ PROBLÈME DÉTECTÉ'}")

🔍 TEST 2: VÉRIFICATION DES VALEURS MANQUANTES

📊 VALEURS MANQUANTES PAR COLONNE CRITIQUE :
   ✅ latitude        :    0 manquantes (0.00%)
   ✅ longitude       :    0 manquantes (0.00%)
   ✅ typologie       :    0 manquantes (0.00%)
   ❌ arrondissement  :  148 manquantes (2.97%)

🎯 RÉSULTAT DU TEST : ❌ PROBLÈME DÉTECTÉ



<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3> Verification de coherance geographique  </h3>



<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3> Verification des categories </h3>



<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3> Verification specifique pour DBSCAN </h3>



<!DOCTYPE html>
<html>
<head>
    <style>
        body { font-family: Arial, sans-serif; margin: 40px; background: #f5f5f5; color: #333; line-height: 1.6; }
        h1 { text-align: center; color: lightgreen; font-size: 36px; font-weight: bold; margin-bottom: 10px; }
        h2 { color: #EF806C; font-size: 28px; font-weight: bold; margin-top: 30px; }
        h3 { color: #00FFFF; font-size: 22px; font-weight: bold; margin-top: 20px;,  }
        ul { margin-left: 20px; }
        li { margin: 8px 0; }
        hr { height: 2px; border: none; background-color: lightgreen; margin: 20px 0; }
    </style>
</head>
<body>
<h3> Rapport final </h3>
